prototype-release

Maddonix opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Expanded media API: video correction (masking, frame removal, processing history, metadata), media management (video/PDF list, detail, stream), segment validation, video examinations, and Hugging Face model onboarding.

* **Bug Fixes**
  * Safer file/path resolution, improved locking and cooldown reporting, GPU→CPU inference fallback, retries and fallbacks for model downloads and anonymization.

* **Documentation**
  * New AI/model/setup/testing/privacy/configuration guides and setup/run instructions.

* **Tests**
  * Broad new test suites for imports, serializers, services, views, stats, and media workflows.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

maxhild committed:
Reintroducing the correct lookup

maxhild committed:
Changes that get the video import working

maxhild committed:
Merge pull request #264 from wg-lux/lookup-video-import

Lookup and video import were changed locally by dev-02

maxhild committed:
Anonymization Workflow improved

maxhild committed:
wrong import

maxhild committed:
mostly data integrity measures in the video import and processing workflow

maxhild committed:
Fallback to original anonymize function in video import

maxhild committed:
validate segments

Maddonix committed:
remove confusing env sourcing, update project dependencies, re-introduce lx-anonymizer dependency

maxhild committed:
Videocorrection models

maxhild committed:
Anonymizer Implementation optimization - COrrection is implemented

maxhild committed:
Merge pull request #270 from wg-lux/maxhild-patch-1 - delete empty docs files

Maxhild patch 1

maxhild committed:
Merge pull request #271 from wg-lux/maxhild-patch-2

Maxhild patch 2

Maddonix committed:
use lx-anonymizer from pypi

Maddonix committed:
rm old submodule

Maddonix committed:
prototype-release

maxhild committed:
Media routes

maxhild committed:
Merge pull request #272 from wg-lux/prototype

Prototype changes merged to apicleanup

maxhild committed:
model not existing fallback

maxhild committed:
rm deprecated function apply anonymized pdf from pdf import workflow

maxhild committed:
New api structure mostly finished

maxhild committed:
Video file streaming problems

coderabbitai[bot] committed:
CodeRabbit Generated Unit Tests: Add tests for models, serializers, services, views and coverage

Maddonix committed:
Merge pull request #274 from wg-lux/coderabbitai/utg/b99a0c6

CodeRabbit Generated Unit Tests: Add tests for models, serializers, services, views and coverage

Maddonix committed:
Merge pull request #273 from wg-lux/apicleanup

Apicleanup by unifying media routes on media/, also including fixes of streaming issues that stemmed from complicated paths in lx-anonymizer and video import

Maddonix committed:
update dependencies, fix first tests; update environment path handling

maxhild committed:
Using functions from anonymize for consitency, update to paths module

maxhild committed:
Merge remote-tracking branch 'refs/remotes/origin/apicleanup' into apicleanup

Maddonix committed:
fix test_video_import.py tests

Maddonix committed:
update comment

Maddonix committed:
Merge pull request #276 from wg-lux/apicleanup

Apicleanup

Maddonix committed:
type hinting

Maddonix committed:
Merge remote-tracking branch 'refs/remotes/origin/prototype' into prototype

maxhild committed:
Update endoreg_db/models/media/video/video_file.py

Update sensitive meta processed later

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

maxhild commented on devenv.nix:
```diff
@@ -70,8 +71,8 @@ in
-    UV_PYTHON = lib.mkForce "${python}/bin/python";
-    UV_PYTHON_DOWNLOADS = "never";
+    # UV_PYTHON = lib.mkForce "${python}/bin/python";
+    # UV_PYTHON_DOWNLOADS = "never";
```
languages.python = {
    enable = true;
    package = python;
    uv = {
      enable = true;
      package = uvPackage;
      sync.enable = true;
    };
  };
  
  
Should be handled by this. From devenv documentation:

Options
languages.python.enable

Whether to enable tools for Python development.

Type: boolean

Default: false

Example: true

Declared by: - https://github.com/cachix/devenv/blob/main/src/modules/languages/python.nix
languages.python.package

The Python package to use.

Type: package

Default: pkgs.python3

Declared by: - https://github.com/cachix/devenv/blob/main/src/modules/languages/python.nix
languages.python.directory

The Python project’s root directory. Defaults to the root of the devenv project. Can be an absolute path or one relative to the root of the devenv project.

Type: string

Default: config.devenv.root

Example: "./directory"

Declared by: - https://github.com/cachix/devenv/blob/main/src/modules/languages/python.nix

maxhild committed:
Update endoreg_db/models/media/video/video_file_io.py

improved monitoring

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

Maddonix committed:
intermediate state, media-tests work

Maddonix committed:
improve ffmpeg executable path resolvement

Maddonix committed:
add copilot instructions

Maddonix committed:
improve video import service

Maddonix committed:
further test fixing

Maddonix committed:
fix view tests

Maddonix committed:
fix caching issue

Maddonix committed:
Improve VideoFile; Improve test-suite

maxhild committed:
Adding caching settings to the environment example file

maxhild committed:
Arguments of frame cleaner now include anonymization strategy

maxhild committed:
Not logging hash

maxhild committed:
Invoking create sensitive file to ensure the video file is moved out of processed

maxhild committed:
öate stage wrapping of anonymize function - coderabbit review

maxhild committed:
versionbump

maxhild committed:
refactor paths

maxhild committed:
version 0.8.2.2 - global env

maxhild committed:
0.8.2.3 - other one existed

maxhild committed:
register correction view

maxhild committed:
version bump

maxhild committed:
Frame cleaner refactor has cleaned up some things

maxhild committed:
correct update after correction

maxhild committed:
rm reimport view

maxhild committed:
parentheses in method call

maxhild committed:
avoid circular model import

maxhild committed:
rm mapping import

maxhild committed:
Paths, Documentation update

maxhild committed:
Merge branch 'prototype' into video_import

maxhild committed:
Merge branch 'prototype' into stuttgart

maxhild committed:
Version 0.8.2.2 - Paths gotten from env

Currently running on lx annotate

maxhild committed:
Merge branch 'prototype' into video_import

Maddonix committed:
Merge pull request #277 from wg-lux/video_import

Frame cleaner refactor, sensitive meta type annotation, cleaned up reimport views

Maddonix committed:
Bump version to 0.8.2.5

Maddonix committed:
Bump version to 0.8.2.5

Maddonix committed:
Actually bump version to 0.8.2.5

Maddonix committed:
Published version of Endoreg-DB 0.8.2.5

Maddonix committed:
bumped version due to lx-anonymizer failed version

maxhild committed:
meta logic

maxhild committed:
version and hf support in pyproject and uv lock

maxhild committed:
Merge branch 'prototype' into model_meta

maxhild committed:
Merge pull request #280 from wg-lux/model_meta

Model meta inferrence from hf

maxhild committed:
Version 0.8.2.9 - had to remove some fallbacks for clarity and processing integrity

maxhild committed:
Version 0.8.3.0 - Using huggingface model in pipe 1

maxhild committed:
Version 0.8.3.0 - Using huggingface model in pipe 1

maxhild committed:
version 0.8.3.1 switched processing order in video import, initializing video object first

maxhild committed:
EndoscopyProcessor  import

Maddonix committed:
create model meta from huggingface command

Maddonix committed:
AI model setup documentation and setup command

Maddonix committed:
management unified setup command changed to use cachetable

Hamzaukw committed:
A global autouse=True fixture was replacing the real Django EndoscopyProcessor model with a dummy, causing .objects lookups to fail in integration tests:Removed autouse=True and explicitly applied the fixture only to the unit tests

Updated test functions to use patch_env fixture for isolated environment setup.

maxhild committed:
0.8.3.7 - Correctly updating processing context in video import

maxhild committed:
setup creating model meta
0.8.3.8

maxhild committed:
create correct model from hvf download when running endoreg_db setup command

Hamzaukw committed:
FAILED tests/media/video/test_video_import.py::test_get_processor_roi_info_returns_valid - assert False........the test monkey-patched EndoscopyProcessor in the models module, but the service had already imported its own copy, so the isinstance check failed; it was resolved by also monkey-patching the service's referenced

Hamzaukw committed:
mock_frame_cleaning:should return two values (line # 88)

Hamzaukw committed:
The real code updates the state returned by get_or_create_state(), but our test dummy creates a new state object every time. So _finalize_processing() updates that temporary state, while the test checks v.state, which never changes so the test fails.

maxhild committed:
hopefully fixes model meta and frame cleaner instantiation

Maddonix committed:
Bump version to 0.8.4.3 - setup of ai model to respect local files and new models, configuration is done via setup conf script and respective model yamls and setup_conf.yaml

Maddonix committed:
rm merge conflict, reapplied instantiation fix

Maddonix committed:
rm merge conflict, reapplied instantiation fix

Maddonix committed:
version 0.8.4.4 - changes from both machines integrated

Maddonix committed:
model meta logic force recreate fix -version 0.8.4.7

Maddonix committed:
Bump version to 0.8.4.8 - debugging and pdf blackening standard

maxhild committed:
Ensuring center is set in video create from file, otherwise setting fallback from .env

maxhild committed:
Version 0.8.4.9 - ensuring center

maxhild committed:
0.8.5.5 Update center from center object not name

maxhild committed:
0.8.5.1 Guards against deleting raw video on same hash

maxhild committed:
version 0.8.5.3 - Accepting strings for center names for env variable loading

maxhild committed:
version 0.8.5.3 - Accepting strings for center names for env variable loading

maxhild committed:
Version 0.8.5.4 General none check in sensitive meta update from dict

maxhild committed:
videoexamination api update

maxhild committed:
version 0.8.5.5 - tests and pdf reimport fix

maxhild committed:
0.8.5.8 - Improved naming and performance of vop_patient_data -> now sensitive_patient_data

maxhild committed:
Removing views and implementing pdf dual stream

maxhild committed:
dual pdf stream, anonym vs raw, fixing cross origin protection problems

maxhild committed:
Lookup now supports tags

Maddonix commented on endoreg_db/management/commands/setup_endoreg_db.py:
```diff
@@ -0,0 +1,381 @@
+                        weights_path = str(dest_path.relative_to(STORAGE_DIR))
+                        self.stdout.write(f"    Copied weights to: {dest_path}")
+
+                # Create basic metadata using configurable defaults
+                meta = ModelMeta.objects.create(
+                    name=model.name,
+                    version="1.0",
+                    model=model,
+                    labelset=labelset,
+                    weights=weights_path,  # Set weights if available
+                    activation=defaults.get("activation", "sigmoid"),
+                    mean=defaults.get("mean", "0.485,0.456,0.406"),
+                    std=defaults.get("std", "0.229,0.224,0.225"),
+                    size_x=defaults.get("size_x", 224),
+                    size_y=defaults.get("size_y", 224),
+                    axes=defaults.get("axes", "CHW"),
+                    batchsize=defaults.get("batchsize", 32),
+                    num_workers=defaults.get("num_workers", 4),
+                    description=f"Auto-generated metadata for {model.name}",
+                )
+
```
@coderabbitai create an issue and assign @maxhild

Maddonix commented on endoreg_db/models/media/pdf/raw_pdf.py:
```diff
@@ -121,13 +144,73 @@ def anonymized_file_path(self) -> Path:
             except (ValueError, AttributeError, NotImplementedError):
                 return None
         return None
-    
+
     def set_anonymized_file_path(self, file_path: Path):
         """
         Sets the file path of the anonymized PDF file.
         """
-        self.anonymized_file = File(file_path)
-        self.save(update_fields=['anonymized_file'])
+        self.anonymized_file = File(file_path)  # type: ignore
+        self.save(update_fields=["anonymized_file"])
```
@coderabbitai create an issue and assign @maxhild

Maddonix commented on endoreg_db/models/media/video/pipe_1.py:
```diff
@@ -61,7 +64,16 @@ def _pipe_1(
                     model_meta = ai_model_obj.get_latest_version()
             except AiModel.DoesNotExist:
                 logger.error(f"Pipe 1 failed: Model '{model_name}' not found.")
-                return False
+                try:
+                    model_name = download_segmentation_model()
+                    ai_model_obj = AiModel.objects.get(name=model_name)
+                    if model_meta_version is not None:
+                        model_meta = ai_model_obj.metadata_versions.get(version=model_meta_version)
+                    else:
+                        model_meta = ai_model_obj.get_latest_version()
+                except AiModel.DoesNotExist:
+                    logger.error(f"Pipe 1 failed: Model '{model_name}' not found.")
+                    return False
```
@coderabbitai create an issue and assign @maxhild

Maddonix commented on endoreg_db/models/media/video/video_file_ai.py:
```diff
@@ -297,8 +297,43 @@ def _predict_video_pipeline(
         logger.error("Inference failed for video %s: %s", video.uuid, e, exc_info=True)
-        # Raise exception
-        raise RuntimeError("Inference failed") from e
+        # CUDA-OOM Fallback: Speicher freigeben und CPU versuchen
+        try:
+            import torch, gc
+            is_oom = isinstance(e, (getattr(torch.cuda, 'OutOfMemoryError', RuntimeError), RuntimeError)) and (
+                'out of memory' in str(e).lower() or 'cuda out of memory' in str(e).lower()
+            )
+        except Exception:
+            is_oom = False
+        if 'torch' in globals() or 'torch' in locals():
+            try:
+                import torch  # ensure available in this scope
+                if torch.cuda.is_available() and is_oom:
+                    logger.warning("CUDA OOM detected. Freeing CUDA cache and retrying on CPU…")
+                    try:
+                        torch.cuda.empty_cache()
+                        gc.collect()
+                    except Exception:
+                        pass
+                    try:
+                        # Move model to CPU and retry inference
+                        _ = ai_model_instance.cpu()
+                        classifier = Classifier(ai_model_instance, verbose=True)
+                        predictions = classifier.pipe(string_paths, crops)
+                        logger.info("Inference completed on CPU after CUDA OOM for video %s.", video.uuid)
+                    except Exception as e2:
+                        logger.error("CPU fallback inference failed for video %s: %s", video.uuid, e2, exc_info=True)
+                        # Raise exception
+                        raise RuntimeError("Inference failed") from e2
+                else:
+                    # Raise exception
+                    raise RuntimeError("Inference failed") from e
+            except Exception:
+                # Raise exception
+                raise RuntimeError("Inference failed") from e
+        else:
+            # Raise exception
+            raise RuntimeError("Inference failed") from e
```
@coderabbitai create an issue and assign me

Maddonix commented on endoreg_db/models/media/video/video_file_frames/_manage_frame_range.py:
```diff
@@ -111,6 +112,17 @@ def _extract_frame_range(
 
         return True
 
+    except FileNotFoundError as err:
+        logger.error(
+            "Frame range extraction [%d, %d) failed for video %s: %s",
+            start_frame,
+            end_frame,
+            video.uuid,
+            err,
+            exc_info=True,
+        )
+        raise
```
@coderabbitai create a follow up issue and assign me

Maddonix committed:
update instruction

Maddonix committed:
major refactoring; video logic; Syntax; Huggingface Model Deployment, update base model pytorch version, migrate to .safetensor

Maddonix committed:
fix video_deletion_validation test logic to properly use default data

Maddonix committed:
reintroduce display_name functionality; reintroduce active_file_url

Maddonix committed:
minor fix after adding user_tags

Maddonix committed:
rough fix to allow .safetensor loading

Maddonix committed:
improve labelset version handling

Maddonix committed:
fix type hinting

Maddonix committed:
deprecate case template based case generator; mirgrate (including migrations from previous commit)

Maddonix committed:
minor type fixing
